### PR TITLE
Add `eigs` specialisation.

### PR DESCRIPTION
**Pull request by Ericgig**
Use the reduce branch for `isherm`. Merge after #8

### CODE REVIEW

**Comment by Ericgig** [comment]
@AGaliciaMartinez , @quantshah I fixed the conflict of this one.
#15 will wait until am back from vacation.

**Comment by AGaliciaMartinez** [comment]
Ofc, enjoy your vacations!

File: src/qutip_jax/linalg.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    N = data.shape[0]
+    if data.shape[0] != data.shape[1]:
+        raise TypeError("Can only diagonalize square matrices")
+    if sort not in ('low', 'high'):
+        raise ValueError("'sort' must be 'low' or 'high'")
+    if eigvals > N:
+        raise ValueError("Number of requested eigen vals/vecs must be <= N.")
+    eigvals = eigvals or N
+    # Let dict raise keyerror of
+    low_first = {"low": True, "high": False}[sort]
```
Comment (AGaliciaMartinez) [inline src/qutip_jax/linalg.py:62]: I suggest we raise `ValueError` and instead as I think is a bit more informative.
```suggestion
    if sort not in ['low', 'high']:
        raise ValueError(f"`sort` must be 'low' or 'high'. Not {sort}.")
    low_first = sort=='low'
```

**Review by AGaliciaMartinez — APPROVED** [review]
Nice work Eric! I left some small suggestions but I think this is ready to be merged.

Perhaps we can add some tests to check the raising errors but I can do this later or we can leave it as good first issue.

File: src/qutip_jax/linalg.py (hunk 1 above)
Comment (Ericgig) [reply src/qutip_jax/linalg.py:61]: It was already raised...

**Review by Ericgig — COMMENTED** [review]
Thank you for the review

**Pull request merged**